### PR TITLE
[ISSUE #38]add nodeAffinity to support scheduling pods to specified nodes

### PR DESCRIPTION
**Pull request by xiangwangcheng**
Use NodeAffinity to specify nodes that wanted.

### CODE REVIEW

File: example/rocketmq_v1alpha1_broker_cr.yaml [hunk 1]
```diff
@@ -27,6 +27,10 @@ spec:
   replicationMode: ASYNC
   # replicaPerGroup is the number of each broker cluster
   replicaPerGroup: 1
+  # nodeAffinityKey|nodeAffinityKey|nodeAffinityValues used to schedule specified nodes
```
Comment (liuruiyiyang) [inline example/rocketmq_v1alpha1_broker_cr.yaml:30]: How should the users write the CR config yaml if they do not need NodeAffinity ?

File: pkg/controller/broker/broker_controller.go [hunk 2]
```diff
@@ -380,6 +380,23 @@ func (r *ReconcileBroker) getBrokerStatefulSet(broker *rocketmqv1alpha1.Broker,
 		statefulSetName = broker.Name + "-" + strconv.Itoa(brokerGroupIndex) + "-replica-" + strconv.Itoa(replicaIndex)
 	}
 
+	var affinity corev1.Affinity
+	if broker.Spec.NodeAffinityKey != "" && broker.Spec.NodeAffinityOperator != "" && broker.Spec.NodeAffinityValues != nil {
```
Comment (liuruiyiyang) [inline pkg/controller/broker/broker_controller.go:384]: Have you tried if the user do not config Affinity and what happens when affinity is nil ?

**Comment by liuruiyiyang** [comment]
Hi, @xiangwangcheng.  Thanks for your contribution :)
The PR seems has no big problem but one question: does the NameServer need Affinity? If so could you please also add Affinity to NameServer CRD? Thanks.

**Comment by caigy** [comment]
The latest version has supported affinity, if problem still exists, please feel free to reopen an issue.

**Pull request closed without merging**